src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,97 @@
-export const Pagination = () => {};
+import { getNumbers, getPaginatedItems } from '../../utils';
+import classNames from 'classnames';
+
+type Props = {
+  total: number;
+  perPage: number;
+  currentPage?: number;
+  setCurrentPage: (arg: number) => void;
+};
+
+export const Pagination = ({
+  total,
+  perPage,
+  currentPage = 1,
+  setCurrentPage,
+}: Props) => {
+  const pages = getNumbers(1, Math.ceil(total / perPage));
+  const items = getNumbers(1, total).map(n => `Item ${n}`);
+  const paginatedItems = getPaginatedItems(items, perPage);
+
+  const isCurrentPageFirst = currentPage === 1;
+  const isCurrentPageLast = currentPage === pages.length;
+
+  const handleBackwardsClick = () => {
+    if (!isCurrentPageFirst) {
+      setCurrentPage(currentPage - 1);
+    }
+  };
+
+  const handleForwardsClick = () => {
+    if (!isCurrentPageLast) {
+      setCurrentPage(currentPage + 1);
+    }
+  };
+
+  return (
+    <>
+      <ul className="pagination">
+        <li
+          className={classNames('page-item', { disabled: isCurrentPageFirst })}
+        >
+          <a
+            data-cy="prevLink"
+            className="page-link"
+            href="#prev"
+            aria-disabled={isCurrentPageFirst}
+            onClick={handleBackwardsClick}
+          >
+            «
+          </a>
+        </li>
+        {pages.map((page, i) => {
+          const pageNumber = i + 1;
+
+          return (
+            <li
+              key={page}
+              className={classNames('page-item', {
+                active: currentPage === pageNumber,
+              })}
+            >
+              <a
+                data-cy="pageLink"
+                className="page-link"
+                href={`#${pageNumber}`}
+                onClick={() => setCurrentPage(pageNumber)}
```

Comment:
Similar to the previous comment, using `href="#pageNumber"` can change the URL fragment which might not be desirable. Consider using `button` elements instead of `a` elements, or prevent default behavior in the click handler.